apps/src/templates/instructions/MarkdownInstructions.jsx

```diff
@@ -53,6 +55,22 @@ const MarkdownInstructions = React.createClass({
       });
     }
 
+    if (this.props.isBlockly) {
+      // Convert any inline XML into blockly blocks. Note that we want to
+      // make sure we don't initialize any blockspace before the main
+      // block space has been created, lest we violate some assumptions
+      // blockly has.
```

Comment:
I wonder if it would be better to give Blockly the responsibility of initializing right away, or after `MAIN_BLOCK_SPACE_CREATED`?  E.g.:

```js
// In this file
Blockly.BlockSpace.createSecondaryBlockSpace(() => {
  convertXmlToBlockly(ReactDOM.findDOMNode(this));
  this.props.onResize();
}
```

```js
// In Blockly
Blockly.BlockSpace.createSecondaryBlockSpace = function (callback) {
  if (Blockly.mainBlockSpace) {
    callback();
  } else {
    document.addEventListener(Blockly.BlockSpace.EVENTS.MAIN_BLOCK_SPACE_CREATED, () => {
      callback();
    });
  }
};
```

Totally up to you!  I don't have great context to know which component should own deferring initialization.